mirrord/config/src/feature/network/incoming.rs

```diff
@@ -267,6 +289,22 @@ pub struct IncomingConfig {
 
     /// #### feature.network.incoming.filter {#feature-network-incoming-http-filter}
     pub http_filter: HttpFilterConfig,
+
+    /// #### feature.network.incoming.listen_ports {#feature-network-incoming-listen_ports}
+    ///
+    /// Mapping for local ports to actually used local ports.
+    /// When application listens on a port while steal/mirror is active
+    /// we fallback to random ports to avoid port conflicts.
+    /// Using this configuration will always use the specified port.
+    /// If this configuration doesn't exist, mirrord will try to listen on the original port
+    /// and if it fails it will assign a random port
+    ///
+    /// This is useful when you want to access ports exposed by your service locally
+    /// For example, if you have a service that listens on port `80` and you want to access it,
+    /// you probably can't listen on `80` without sudo, so you can use `[[80, 4480]]`
+    /// then access it on `4480` while getting traffic from remote `80`.
+    /// The value of `port_mapping` doesn't affect this.
+    pub listen_ports: BiMap<u16, u16>,
```

Comment:
With `BiMap` the user can't have an invalid config such as `[[10, 1234], [10, 1234]]`, right?

I'm assuming this holds true for the other `listen_ports`?

Reply:
yeah